src/main/java/com/example/crm_gym/controllers/TrainerController.java

```diff
@@ -0,0 +1,250 @@
+package com.example.crm_gym.controllers;
+
+import com.example.crm_gym.dto.*;
+import com.example.crm_gym.logger.TransactionLogger;
+import com.example.crm_gym.models.*;
+import com.example.crm_gym.services.TrainerService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RestController
+@Slf4j
+@RequestMapping("/trainers")
+@Api(produces = "application/json", value = "Operations for creating, updating, retrieving and deleting trainers in the application")
+public class TrainerController {
+
+    private final TrainerService trainerService;
+
+    @Autowired
+    public TrainerController(TrainerService trainerService) {
+        this.trainerService = trainerService;
+    }
+
+    @PostMapping("/register")
```

Comment:
remove everything this verbs from path. You are able to use POST for creating(registering)  PUT for updating and you don't need to point additional actions like/register /update
We have exceptional case  - It is controller
https://github.com/mjc-school/MJC-School/blob/main/stage%20%233/module%20%234.%20REST%20Basics%20-%20Spring%20Web/lesson%20%231.%20REST%20Basics/content.md#controller